drivers/sx1280/sx1280.c

```diff
@@ -0,0 +1,413 @@
+    case LORA_BW_800_KHZ:
+        return RAL_LORA_BW_800_KHZ;
+    case LORA_BW_1600_KHZ:
+        return RAL_LORA_BW_1600_KHZ;
+    default:
+        return -1;
+    }
+}
+
+static ral_lora_sf_t _lora_sf_to_ral_sf(int sf)
+{
+    return (ral_lora_sf_t)sf;
+}
+
+static ral_lora_cr_t _lora_cr_to_ral_cr(int cr)
+{
+    switch (cr) {
+    case LORA_CR_4_5:
+        return RAL_LORA_CR_4_5;
+    case LORA_CR_4_6:
+        return RAL_LORA_CR_4_6;
+    case LORA_CR_4_7:
+        return RAL_LORA_CR_4_7;
+    case LORA_CR_4_8:
+        return RAL_LORA_CR_4_8;
+    case LORA_CR_LI_4_5:
+        return RAL_LORA_CR_LI_4_5;
+    case LORA_CR_LI_4_6:
+        return RAL_LORA_CR_LI_4_6;
+    case LORA_CR_LI_4_8:
+        return RAL_LORA_CR_LI_4_8;
+    default:
+        return -1;
+    }
+}
+
+static uint32_t _get_bw(ral_lora_bw_t lora_bw)
+{
+    switch (lora_bw) {
+    case RAL_LORA_BW_007_KHZ:
+        return (uint32_t)7000;
+    case RAL_LORA_BW_010_KHZ:
+        return (uint32_t)10000;
+    case RAL_LORA_BW_015_KHZ:
+        return (uint32_t)15000;
+    case RAL_LORA_BW_020_KHZ:
+        return (uint32_t)20000;
+    case RAL_LORA_BW_031_KHZ:
+        return (uint32_t)31000;
+    case RAL_LORA_BW_041_KHZ:
+        return (uint32_t)41000;
+    case RAL_LORA_BW_062_KHZ:
+        return (uint32_t)62000;
+    case RAL_LORA_BW_125_KHZ:
+        return (uint32_t)125000;
+    case RAL_LORA_BW_200_KHZ:
+        return (uint32_t)200000;
+    case RAL_LORA_BW_250_KHZ:
+        return (uint32_t)250000;
+    case RAL_LORA_BW_400_KHZ:
+        return (uint32_t)400000;
+    case RAL_LORA_BW_500_KHZ:
+        return (uint32_t)500000;
+    case RAL_LORA_BW_800_KHZ:
+        return (uint32_t)800000;
+    case RAL_LORA_BW_1600_KHZ:
+        return (uint32_t)1600000;
+    default:
+        return 0;
+    }
+}
+
+static void _dio1_isr(void *arg)
+{
+    netdev_trigger_event_isr(arg);
+}
+
+static int enable_irq(const sx1280_t *dev)
+{
+    const uint16_t irq_mask = (
+        SX1280_IRQ_TX_DONE |
+        SX1280_IRQ_RX_DONE |
+        SX1280_IRQ_PBL_DET |
+        SX1280_IRQ_HEADER_VALID |
+        SX1280_IRQ_HEADER_ERROR |
+        SX1280_IRQ_CAD_DONE |
+        SX1280_IRQ_CAD_DET |
+        SX1280_IRQ_TIMEOUT
+        );
+
+    int res = sx1280_set_dio_irq_params(dev, irq_mask, irq_mask, SX1280_IRQ_NONE, SX1280_IRQ_NONE);
+
+    return res;
+}
+
+uint32_t sx1280_random(const sx1280_t *dev)
+{
+    uint32_t rnd = 0;
+
+    /* Disable IRQ */
+    int res = 0;
+
+    res = sx1280_set_dio_irq_params(dev, SX1280_IRQ_NONE, SX1280_IRQ_NONE, SX1280_IRQ_NONE,
+                                    SX1280_IRQ_NONE);
+    if (res != SX1280_STATUS_OK) {
+        DEBUG("[sx1280] init: ERROR failed irq disabled, %d\n", res);
+        return -EIO;
+    }
+
+    sx1280_set_rx(dev->ral.context, SX1280_TICK_SIZE_1000_US, SX1280_RX_CONTINUOUS_MODE );
+    int16_t rssi = 0;
+
+    for (unsigned i = 0; i < 32; i++) {
+        ztimer_sleep(ZTIMER_MSEC, 1);   /* wait one millisecond */
```

Comment:
Why 1ms? Is there a define somewhere?

Reply:
@Aymeric-Brochier mentioned offline that this was copied from `sx126x/sx127x` code, but from the `sx127x` datasheet I can see that the register there is valid to use for random number generation, I don't see such mention in the `sx1280` datasheet, lets remove those functions then IMO